pilot/pkg/security/trustdomain/bundle.go

```diff
@@ -50,6 +50,10 @@ func NewTrustDomainBundle(trustDomain string, trustDomainAliases []string) Bundl
 // If the local trust domain is "td2" and its alias is "td1" (migrating from td1 to td2),
 // replaceTrustDomainAliases returns ["td2/ns/foo/sa/bar", "td1/ns/foo/sa/bar]].
 func (t Bundle) ReplaceTrustDomainAliases(principals []string) []string {
+	if len(t.TrustDomains) == 0 {
```

Comment:
I don't think this will ever happen. If the trust domain is not set, Istio will use "cluster.local" as the default value.

Reply:
Actually I just realized this line doesn't fix the issue, when calling from https://github.com/istio/istio/blob/0297cf12b5bb686bce0ca25668e699c958977b52/pilot/pkg/security/authz/converter/converter.go#L232, it passes an empty string as trust domain.

Reply:
Updated to pass `cluster.local` by default.